test/integration/index.spec.js

```diff
@@ -184,7 +201,38 @@ describe('/functions', () => {
 
     assert.deepStrictEqual(functionById, functionFiltered)
   })
+  it('POST', async () => {
+    const { data: newFunc } = await axios.post(`${URL}/functions`, func)
+    assert.equal(newFunc.name, 'test_func')
+    assert.equal(newFunc.schema, 'public')
+    assert.equal(newFunc.language, 'sql')
+    assert.equal(newFunc.return_type, 'int4')
+    func.id = newFunc.id
+  })
+  it('PATCH', async () => {
+    const updates = {
+      name: 'test_func_renamed',
+      params: ['integer', 'integer'],
+      schema: 'test_schema',
+      // extension: 'mathlib', // TODO: test patching function extension
+    }
+
+    let { data: updated } = await axios.patch(`${URL}/functions/${func.id}`, updates)
+    assert.equal(updated.id, func.id)
+    assert.equal(updated.name, 'test_func_renamed')
+    assert.equal(updated.schema, 'test_schema')
+  })
+  it('DELETE', async () => {
+    await axios.delete(`${URL}/functions/${func.id}`)
+    const { data: functions } = await axios.get(`${URL}/functions`)
+    const stillExists = functions.some((x) => func.id === x.id)
+    assert.equal(stillExists, false, 'Function is deleted')
+    await axios.post(`${URL}/query`, {
+      query: `DROP SCHEMA test_schema;`,
+    })
```

Comment:
@i-pip i recommend moving this to an `after` hook. See: https://github.com/supabase/postgres-meta/blob/f8757bfa295591f010f6f883ba72b8e7a0b36647/test/integration/index.spec.js#L861